sbu-kalm/KALM
# Api setup

**Updates**
1. Implemented backend APIs for querying for list of frames & adding frames
2. Implemented backend APIS for adding, updating, deleting, and getting roles based off a frame object id
3. Added front-end integrations with tests for backend apis. NOTE - Still need to fully integrate the API to the rest of the buttons on the Front-end. The only working button is the add frame button & test API button. Test API button will be removed once I fully integrate the frontend & backend.

**Testing - Check off with x to verify you've completed the below**

**How to review:**
1. Make sure you have a .env file in the server folder with the appropriate MONGO_DB_CONNECTION_STRING 
```
git checkout main
git fetch
git checkout apiSetup

cd server 
flask run

cd client
npm start
```

1. Go to browser & click Frame Management to see the below. The table should automatically fetch and populate from the database in MongoDB. 
<img width="1627" alt="image" src="https://github.com/sbu-kalm/KALM/assets/53535722/97bb985e-d020-438e-b2ea-77f47dc94f37">
3. Add a frame button should work, fill in with values and see it appear
4. To test backend apis for roles, uncomment each of the sample testing made in FrameManagement.tsx and then click the Test Api Button

![image](https://github.com/sbu-kalm/KALM/assets/53535722/050c68d4-276a-484c-aeb5-17f9f5879e10)
